aodndata/aatams_sattag_qc_dm/aatams_sattag_qc_dm_schema.py

```diff
@@ -0,0 +1,685 @@
+    return afloat if valid else False
+
+
+def not_applicable(astr):  # TODO: handle every case as particular?
+    """Check string for emptiness, not applicable or NaN string.
+
+    Args:
+      astr: a string
+
+    Returns:
+      str: "" if valid
+      bool: False if invalid
+
+    """
+    if astr in ("", "NA", "NaN"):
+        return ""
+    else:
+        return False
+
+
+# Transform functions -> Return mutations
+
+
+def str2list(astr, delimiter):
+    """Split string by a delimiter.
+
+    Args:
+      astr: a string
+      delimiter: a string delimiter
+
+    Returns:
+      list: A list of str splits
+
+    """
+    return astr.split(delimiter)
+
+
+def iter_int(alist):
+    """Transform every list item into integer.
+
+    Args:
+      alist: A list
+
+    Returns:
+      list: a list of integers
+
+    Raises:
+      ValueError: if an item cannot be an int.
+
+    """
+    [int(x) for x in alist]
+
+
+def iter_float(alist):
+    """Transform every list item into float.
+
+    Args:
+      alist: A list
+
+    Returns:
+      list: a list of floats
+
+    Raises:
+      ValueError: if an item cannot be an float.
+
+    """
+    [float(x) for x in alist]
+
+
+def str2date(astr, fmt):
+    """Convert a string to datetime object.
+
+    Args:
+      astr: A string
+      fmt: A datetime.strptime format string
+
+    Returns:
+      datetime: the datetime representation
+
+    """
+    try:
+        return datetime.strptime(astr, fmt)
+    except ValueError:
+        return datetime.strptime(astr.replace("24:00:00", "00:00:00"), fmt)
+
+
+# check functions -> return bool or false
+def check_len_list(alen, alist):
+    """Check the length of a list.
+
+    Args:
+      alen: list len
+      alist: a list
+
+    Returns:
+      bool: True or False
+
+    """
+    return alen == len(alist)
+
+
+def check_file_names(file_names, flist):
+    """Check if filenames are present in a full path file list.
+
+    Args:
+      file_names: a list of file names
+      flist: a list of full path file names
+
+    Returns:
+      bool: True or False
+
+    """
+    ngroup0 = set(file_names)
+    ngroup1 = set([os.path.basename(x).split("_")[0] for x in flist])
+    return ngroup0 == ngroup1
+
+
+def is_positive(anumber):
+    """Check positiveness.
+
+    Args:
+      anumber: a number
+
+    Returns:
+      bool: True or False
+
+    """
+    return anumber > 0
+
+
+def is_negative(anumber):
+    """Check negativeness.
+
+    Args:
+      anumber: a number
+
+    Returns:
+      bool: True or False
+
+    """
+    return anumber < 0
+
+
+def is_equal(x, y):
+    """Check equality between two items.
+
+    Args:
+      x: something
+      y: something else
+
+    Returns:
+      bool: True or False
+
+    """
+    return True if x == y else False
+
+
+check_number_of_files_is_correct = partial(
+    check_len_list, AATAMS_DM_NUMBER_OF_FILES_IN_ZIP
+)
+check_name_of_files_is_correct = partial(check_file_names, AATAMS_DM_FILE_TYPE_NAMES)
+
```

Comment:
This is so to reduce arity - this way, all schema definitions are functions of a single argument - the field being validated.

Reply:
yeah sure, I got that, but my point is why creating check_len_list and check_file_names  with 2 arguments in the first place

Reply:
don't take this comment as something you gotta change. it's fine, it's working, I'm just questioning the usefulness of doing so

Reply:
Ahh - it's because I didn't knew, before-hand, If I would need to reuse the functions to create others - just a composition thing. It's also useful to build like this if I may need to split/move some of the functions to another files (e.g. re-use). This way the port is easy since the functions are independent of "globals".